### ISSUE TITLE
The component doesn't re-render to show the updated value of a counter after the first increment

### ISSUE THREAD

**Issue by pkrinesh** [LegendApp/legend-state#88]
The component doesn't re-render to show the updated value of a counter after the first increment when using the global store but works fine on subsequent increments and this only happens when the initial value is set to 0, otherwise works just fine. link to sandbox https://codesandbox.io/s/legend-state-9b6pln?file=/src/App.js.

Am I missing something here?

```js

import { observable } from "@legendapp/state";
import { observer } from "@legendapp/state/react";

const store = observable({
  legend: {
    count: 0
  }
});

const App = observer(function App() {
  const count = store.legend.count.get();

  return (
    <div className="App">
      <p>{count}</p>
      <button onClick={() => store.legend.count.set((c) => c + 1)}>add</button>
    </div>
  );
});
```

**Comment by jmeistrich** [comment]
That is a bug! I've just written a test for it and fixed it, and will release an update tomorrow (it's 2am here in New Zealand). It's caused by poor ordering of operations in the set function where it's checking whether it's a primitive or not before it computes the function (so it thinks it's not a primitive because it's a function).

**Comment by pkrinesh** [comment]
Okay got it. thanks

**Comment by jmeistrich** [comment]
Ok, fixed in versions `0.23.2` and `1.0-rc.1`. Thanks for the report!
